File: src/practice/digvijay.c
#include <stdio.h>
#include <stdlib.h>
int main(void){
    int terms;
    scanf("%d", &terms);
    while(terms--){
        int base, mod;
        scanf("%d%d", &base, &mod);
        char *ch = (char *)calloc(1000*sizeof(char), sizeof(char));
        scanf("%s", ch);
        int i=0;
        int num = 0;
        while(ch[i] != '\0'){
            int digit = ch[i] - '0';
            num = num * base + digit;
            printf("%d ", num%mod);
            i++;
        }
        printf("\n");
    }
}

File: workspace/digvijay.java
import java.util.Scanner;
class Digvijay {
    public static void main(String[] args) throws Exception {
        Scanner in = new Scanner( System.in );
        int terms = in.nextInt();
        for(int q=0; q<terms; q++){
            int base = in.nextInt();
            int mod = in.nextInt();
            String s = in.next();
            int l = s.length();
            int num = 0;
            int i;
            for(i=0;i<l;i++){
                int digit = s.charAt(i) - '0';
                num = num * base + digit;
                System.out.print(num%mod + " ");
            }
            System.out.println();
        }
    }
}
